Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/ComponentAdapter.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/UI/DocumentPropertyEditor/ComponentAdapter.h>
+#include <QtCore/QTimer>
+
+namespace AZ::DocumentPropertyEditor
+{
+    ComponentAdapter::ComponentAdapter()
+    {
+    }
+
+    ComponentAdapter::ComponentAdapter(AZ::Component* componentInstace)
+    {
+        SetComponent(componentInstace);
+    }
+
+    ComponentAdapter::~ComponentAdapter() = default;
+
+    void ComponentAdapter::OnEntityComponentPropertyChanged(AZ::ComponentId componentId)
+    {
+        if (m_componentInstance->GetId() == componentId)
+        {
+            m_refreshQueue++;
+            QTimer::singleShot(
+                0,
+                [this]()
+                {
+                    DoRefresh();
+                });
+        }
+    }
+
+    void ComponentAdapter::InvalidatePropertyDisplay([[maybe_unused]] AzToolsFramework::PropertyModificationRefreshLevel level)
+    {
+        m_refreshQueue++;
+        QTimer::singleShot(
+            0,
+            [this]()
+            {
+                DoRefresh();
+            });
+    }
+
+    void ComponentAdapter::SetComponent(AZ::Component* componentInstance)
+    {
+        m_componentInstance = componentInstance;
+        AzToolsFramework::PropertyEditorEntityChangeNotificationBus::MultiHandler::BusConnect(m_componentInstance->GetEntityId());
+        AzToolsFramework::ToolsApplicationEvents::Bus::Handler::BusConnect();
```

Comment:
Do you need to disconnect from these buses on destruction? If so, remember to do so in reverse order of connection.

Reply:
Thanks, fixed and added the disconnects :)